### ISSUE TITLE
goNextPageAutomatic not triggering validation errors

### ISSUE THREAD

**Issue by TonisOrmisson** [surveyjs/survey-library#2161]
### Are you requesting a feature, reporting a bug or asking a question?
Its a bug :beetle: :mosquito: :bug: 

### What is the current behavior?
a survey with `goNextPageAutomatic =true` will not show validation errors on value change/enter. 
The progress, bar updates, next page is not displayed, also no error displayed although validation fails.

Using enter here: notice question going green on enter
```
"validators":[{"type":"numeric","minValue":1905,"maxValue":2005}]}]}
```

![bad](https://user-images.githubusercontent.com/6357451/81402772-b2ac7a00-913a-11ea-824c-8c6ecfd90ed4.gif)

### What is the expected behavior?

Validation error should be displayed. Progress not updated.

**Comment by andrewtelnov** [comment]
@TonisOrmisson Please set the following survey property: `checkErrorsMode: "onValueChanged"`.
Here is [the example](https://plnkr.co/edit/xFoNwUV8fIjXO2zd).
We check if there an error or not if goNextPageAutomatic set to true. If there is no errors on the page, we go to the next page. If there is an error, by default we do not show it, unless the checkErrorsMode is "onValueChanged".

Thank you,
Andrew

**Comment by TonisOrmisson** [comment]
an example:
https://plnkr.co/edit/2qg4K2r3CCovUTCn

@andrewtelnov yes, using the suggested  checkErrorsMode is "onValueChanged" will change the behaviour to what one would expect. 

But I think it still feels like a bug. 

Fill out the form. first pages nicely move to next page if answer is given. 
Fill the third page with an invalid value. Lets say soemthing > 2005. 
What happens? Nothing. New page is not rendered? 

Why?
The progress bar (in my original version) gets updated. 
The question background changes indicating that it has been answered (correctly)?
Why the next page is not shown?

Because of validation? Not seeing any :)

setting  checkErrorsMode is "onValueChanged" is a workaround. At that point I should be already on checkErrorsMode: onNextPage since I should be automatically moved to next page. But Neither happens. Im not moved to next page, nor the errors get checked.

**Comment by andrewtelnov** [comment]
@andrewtelnov If we change the behavior then the checkErrorsMode property will start to work incorrectly. It sets to check the error on "onNextPage", but it starts to check "onValueChanged". It would be incorrect behavior.

Right now, our logic is the following, I mean the default behavior. An end-user answers questions on the current page, if  goNextPageAutomatic is true, then on changing a value in single inputs (it doesn't work for matrices or checkboxes), SurveyJS checks, do we have errors on the page? If there is no errors then we go on the next page, if there is an error, then we stay on the page, but do nothing.

I agree, that it is confusing, if there is only one question per page. We are checking can we go on the next page on value changed event, but show errors on next page event.
However, changing the behavior will introduce an implicite logic and it would be even hard to explain.

> The progress bar (in my original version) gets updated.
> The question background changes indicating that it has been answered (correctly)?
> 
This is the default behavior for the default checkErrorsMode. An end-user answers all questions on the page, sees the progress, press the "Next" button, sees that there are errors, corrects them and go further.
Not showing progress, without showing errors (checkErrorMode is onNextPage) could confuse the end-users even more. 

Regarding UI, I would agree, that is not perfect. However, I would suggest to change the checkErrorMode as suggested or/and add a description into your question: 
```
                {
                    type: "text",
                    name: "v3",
                    title: "What is your age?",
                    description: "We expect value from 1905 till 2005",
                    isRequired: true,
                    validators:[
                        {
                            type:"numeric",
                            minValue:1905,
                            maxValue:2005
                        }
                    ]
                }
```
In this case your end-user will not guess and if he/she sees something is wrong, why we are still on the same page, he will read the description and/or sees the error.

Thank you,
Andrew

**Comment by TonisOrmisson** [comment]
@andrewtelnov Thank you for the explanation.

> SurveyJS checks, do we have errors on the page? If there is no errors then we go on the next page, if there is an error, then we stay on the page, but do nothing.

This is something that makes this issue feel like a "bug" to me. Ok, I understand you say it is kind of difficult to explain.

I would say that if the survey is set up to go to next page automatically, it would be logical to display errors if there is a case when otherwise (in case of no errors) the next page would be displayed. If displaying the next page is the expected behavior and next page is not displayed while only validation keeps it from going further, then why not display errors? 

What you describe here:
> SurveyJS checks, do we have errors on the page? If there is no errors then we go on the next page, if there is an error, then we stay on the page, but do nothing.

Why this is not considered as  "onNextPage" event for error checking? 

I'm sorry for being suborn :smile:

**Comment by andrewtelnov** [comment]
@TonisOrmisson I totally understand where you come from. There is the issue, no doubt about it.
OK, let me think about this for couple days. There are several scenarios that are not clear for me right now, for example when you have more then one question on the page. Anyway, I will come back here.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@TonisOrmisson OK. I have changed the behavior one more time :) Added couple new user cases into a new unit test. Your scenario will work as you expected and I hope nobody will complain about this change ;-)
The change will be available in [v1.7.8](https://surveyjs.io/WhatsNew#v1.7.8).

Thank you,
Andrew

**Comment by TonisOrmisson** [comment]
:tada: cool!
